### DISCUSSION TITLE
Identify unnecessary entries of the target platform?

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-pde/eclipse.pde#929] [General]
If there an efficient way to find libraries in the target platform which are not necessary in workspace build / Tycho build?

My process is to remove them from the target platform, Reload it, and see if everything still compiles.

And as error markers in a product configuration file do not react to changes in the target file (IIRC this bug was reported in Bugzilla and never got any attention), I also typically also restart Eclipse which shows me all errors correctly.

**Comment by mickaelistria** [comment]
I don't think it's possible to evaluate whether all deps in a .target are useful, because neither the .target file nor PDE has knowledge of how this would be consumed. A .target that is shared externally cannot know whether all its deps are consumed or not.

**Reply by vogella** [reply]
What about a target file which is not shared and only used for a given set of plug-ins in the workspace?

**Reply by laeubi** [reply]
I don't think PDE has something like that but it would be possible to implement of course. But more interesting would be something that "unused code" tools perform, tell you what is a dependency that has only one or very less references or what is heavily used (so it might be risky to update/upgrade).

Another interesting part is if one can *minimize* the target, e.g mention less root IUs because the are already transitive required.

I think the API/Information is there, so one needs a PR to add such buttons to the target editor.

**Reply by vogella** [reply]
Thanks @laeubi for confirming that this is currently not out-of-the-box available.

**Reply by HannesWell** [reply]
I think this is really similar to detecting duplicates/unnessecary elements in products and features.
If done with the right abstraction the same algorithm could be used for all of these three cases.
